pkg/gitutils/manifests.go

```diff
@@ -79,70 +68,12 @@ func CopyManifests(wt *gogit.Worktree, fs embed.FS, root string, mgmtPath string
 func CopyPatchManifests(wt *gogit.Worktree, filePath string, clusterPath string,
 	baseRepoUrl string, baseRepoPath string, baseRepoRevision string) error {
 	log := log.GetLogger()
-	files, err := ioutil.ReadDir(filePath)
+	src, err := os.OpenFile(filePath, os.O_RDONLY, os.ModePerm)
 	if err != nil {
-		fmt.Println(err)
-		return fmt.Errorf("Failed to read the directory %s", err)
-	}
-	var targetData []target
-	for _, file := range files {
-		newFilePath := path.Join(filePath, file.Name())
-		src, err := os.OpenFile(newFilePath, os.O_RDONLY, os.ModePerm)
-		if err != nil {
-			_ = src.Close()
-			return fmt.Errorf("failed to open patch file %s: %s", filePath, err)
-		}
-
-		kustomFile, err := ioutil.ReadFile(newFilePath)
-		if err != nil {
-			_ = src.Close()
-			return fmt.Errorf("Failed to read the patch file %s", err)
-		}
-		parsedData := make(map[interface{}]interface{})
-
-		err = yaml.Unmarshal(kustomFile, &parsedData)
-		if err != nil {
-			_ = src.Close()
-			return fmt.Errorf("Failed to load the parsed data %s", err)
-		}
-		var targetComp []string
-		var kind string
-		var information info
-		for kustomKey, kustomVal := range parsedData {
-			if kustomKey == "apiVersion" {
-				strKustomVal := fmt.Sprintf("%v", kustomVal)
-				targetComp = strings.Split(string(strKustomVal), "/")
-			}
-			if kustomKey == "kind" {
-				kind = fmt.Sprintf("%v", kustomVal)
-			}
-		}
-		information = info{
-			Group:   string(targetComp[0]),
-			Version: string(targetComp[1]),
-			Kind:    kind,
-		}
-		targetData = append(targetData, target{
-			information, file.Name(),
-		})
-
-		// remove manifests/ prefix
-		components := strings.Split(newFilePath, "/")
-		dstPath := path.Join(components[len(components)-1])
-		dstPath = path.Join(clusterPath, dstPath)
-		dst, err := wt.Filesystem.Create(dstPath)
-		if err != nil {
-			_ = src.Close()
-			return fmt.Errorf("failed to create destination file %s: %s", dstPath, err)
-		}
-		_, err = io.Copy(dst, src)
-		_ = src.Close()
-		_ = dst.Close()
-		if err != nil {
-			return fmt.Errorf("failed to copy embedded file: %s", err)
-		}
-		log.V(1).Info("copied embedded file", "destination", dstPath)
+		return fmt.Errorf("failed to open the patch file %s:", err)
 	}
+	components := strings.Split(filePath, "/")
+	dstPath := path.Join(components[len(components)-1])
```

Comment:
It's not obvious from the code, but it looks like you're trying to get the file name component, which is the last component of the path. Is that correct? If so, I have a few questions/comments:
- Why use `path.Join()`? There's only one component here.
- A better name would be fileName
- Instead of strings.Split(), consider using path.Split(). Read the docs, I think it extracts the file name.

Reply:
Yes, I've made these changes to the code. I started using filepath.Split as well